js/geo/transform.js

```diff
@@ -186,16 +179,16 @@ class Transform {
     zoomScale(zoom) { return Math.pow(2, zoom); }
     scaleZoom(scale) { return Math.log(scale) / Math.LN2; }
 
-    project(lnglat, worldSize) {
```

Comment:
These parameters are not related to supporting different tile sizes. I think I originally introduced it when implementing `flyTo` (which was the only occurrence where the `worldSize` argument was used outside of Transform), but I found a way to get rid of it. 

The only transform method that cares about source tile sizes is `coveringZoomLevel`, which I think should be moved elsewhere. Moreover,`Transform` `tileSize` doesn't really have the meaning it implies by its name — what it really does is establish an arbitrary relationship between zoom value and the pixel width of the world, so that e.g. it's 512px when we set zoom level to 0. I'd like to refactor this to make things clearer.